xapian-core/tests/api_db.cc

```diff
@@ -1687,14 +1687,54 @@ DEFINE_TESTCASE(sortrel1, backend) {
     return true;
 }
 
-// Test network stats and local stats give the same results.
-DEFINE_TESTCASE(netstats1, remote) {
-    XFAIL_FOR_BACKEND("multi_glass_remoteprog_glass",
-		      "Multi remote databases are currently buggy");
+static void
+make_netstats1_db(Xapian::WritableDatabase& db, const string&)
+{
+    struct { Xapian::docid did; const string text; } content[] = {
+	{1, "This is a test document used with the API test. "
+	    "This paragraph must be at least three lines (including the blank "
+	    "line) to be counted as a \"paragraph\"."},
+	{2, "This is a second simple data test, used to test multiple "
+	    "(inmemory anyway) databases.  The text in this file is"
+	    " unimportant, although I suppose it ought to include "
+	    "the standard word \"word\" in a few places."},
+	{3, "This file will be indexed by paragraph, and the simple query "
+	    "will search for the word \"word\".  Well expect the mset to"
+	    " contain two documents, including this paragraph and the fourth"
+	    ", below.  Since this paragraph uses the word \"word\" so much,"
+	    " this should be the first one in the match set.  Ill just say "
+	    "the word a few more times (word!) to make sure of that.  "
+	    "If this doesnt word (typo, I meant work), then there "
+	    "may be fourletter words spoken."},
+	{4, "Ill leave this at two paragraphs.  This one hasnt got any"
+	    " useful information in it either."},
+	{5, "This paragraph only has a load of absolute rubbish, "
+	    "and nothing of any use whatsoever."},
+	{7, "This is the other paragraph with the word in the simple"
+	    " query in it.  For simplicity, all paragraphs are at least "
+	    "two lines, due to how the hacked up indexer works."},
+	{9, "This is another paragraph which wont be returned.  Well, "
+	    "not with the simple query, anyway."},
+	{11, "And yet another.  This one does mention banana splits, "
+	    "though, so cant be that bad."}
+    };
 
-    XFAIL_FOR_BACKEND("multi_remoteprog_glass",
-		      "Multi remote databases are currently buggy");
+    Xapian::TermGenerator indexer;
+    indexer.set_stemmer(Xapian::Stem("english"));
+    indexer.set_stemming_strategy(indexer.STEM_ALL);
 
+    for (auto& i : content) {
+	Xapian::Document doc;
+	indexer.set_document(doc);
+	indexer.index_text(i.text);
+	db.replace_document(i.did, doc);
+    }
+
+    db.commit();
+}
+
+// Test network stats and local stats give the same results.
+DEFINE_TESTCASE(netstats1, remote) {
```

Comment:
We should run this testcase for `generated` now - otherwise we aren't testing the "all local" case.